internal/infrastructure/kubernetes/deployment.go

```diff
@@ -242,6 +242,24 @@ func (i *Infra) createDeployment(ctx context.Context, infra *ir.Infra) (*appsv1.
 	return expected, nil
 }
 
+// deleteService deletes the Envoy Deployment in the kube api server, if it exists.
+func (i *Infra) deleteDeployment(ctx context.Context) error {
+	deploy := new(appsv1.Deployment)
+	key := types.NamespacedName{
+		Namespace: i.Namespace,
+		Name:      envoyDeploymentName,
+	}
+	err := i.Client.Get(ctx, key, deploy)
```

Comment:
thoughts on simplifying this to just calling `i.Client.Delete` ? if I had to guess, the server would return with a similar non found error, in case the deployment doesn't exist